napari/_qt/qt_dims.py

```diff
@@ -85,33 +87,42 @@ def nsliders(self):
         """
         return len(self.sliders)
 
-    def _update_slider(self, slider_index: int):
+    def _update_slider(self, axis: int):
         """
-        Updates everything for a given slider
+        Updates everything for a given slider.
 
         Parameters
         ----------
-        slider_index : slider index (corresponds to axis index)
+        axis : int
+            Axis index.
         """
-        if slider_index >= self.nsliders:
+        # print('range', range, slider_index, self._slider_axis)
+
+        if axis not in self._slider_axis:
             return
+        slider_index = self._slider_axis.index(axis)
 
         slider = self.sliders[slider_index]
 
-        mode = self.dims.mode[slider_index]
+        mode = self.dims.mode[axis]
         if mode == DimsMode.POINT:
             slider.collapse()
-            slider.setValue(self.dims.point[slider_index])
+            slider.setValue(self.dims.point[axis])
         elif mode == DimsMode.INTERVAL:
             slider.expand()
-            slider.setValues(self.dims.interval[slider_index])
+            slider.setValues(self.dims.interval[axis])
         # Set the maximum values of the range slider to be one step less than
         # the range of the layer as otherwise the slider can move beyond the
         # shape of the layer as the endpoint is included
-        range = self.dims.range[slider_index]
+        range = self.dims.range[axis]
         range = (range[0], range[1] - range[2], range[2])
         if range not in (None, (None, None, None)):
-            slider.setRange(range)
+            if range[1] == 0:
```

Comment:
i feel like a better fix would be not creating the sliders in the first place but i guess this is fine for now

Reply:
unfortunately the first time we create them right now we actually don't know the range that will end up getting set. agreed it would be cleaner, but lets revisit when we get to the dims swapping